torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -475,10 +475,11 @@ void ScriptModuleSerializer::writeArchive(
       [&](const at::Tensor& tensor) {
         // returns a string to use in picker.cpp as storage obj key
         if (tensor_cdata_naming_scheme) {
-          tensor_names.push_back(
+          std::string string_id =
               std::to_string(reinterpret_cast<std::intptr_t>(
-                  tensor.storage().unsafeGetStorageImpl())) +
-              ".storage");
+                  tensor.storage().unsafeGetStorageImpl()));
+          tensor_names.push_back(string_id + ".storage");
+          storage_context_.addStorage(string_id, tensor.storage());
```

Comment:
I feel it's a great idea to use StorageContext. The storage_context_ lives in the life span of the serializer so it should be safe for any tensor serialization. @cccclai could you test if it works for bytecode v5 serialization with quantized models? It looks different than #59488 that you've already tested.

Reply:
Test it by re-introduce #58641 and use the following code to test:

```
for i in range(20):
    print(i)
    m = torch.jit.load(model_path)
    m._save_for_lite_interpreter(model_resave_path)
    mm = _load_for_lite_interpreter(model_resave_path)
```
![image](https://user-images.githubusercontent.com/16430979/121267234-bf852b00-c870-11eb-9dd4-8ca7cbf96c8e.png)